sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -522,8 +532,65 @@ mod checked {
                     pt,
                 )
             }
-            TransactionKind::AuthenticatorStateUpdate(_) => {
-                todo!()
+            TransactionKind::EndOfEpochTransaction(txns) => {
+                for tx in txns {
+                    match tx {
+                        EndOfEpochTransactionKind::ChangeEpoch(change_epoch) => {
+                            advance_epoch(
+                                change_epoch,
+                                temporary_store,
+                                tx_ctx,
+                                move_vm,
+                                gas_charger,
+                                protocol_config,
+                                metrics.clone(),
+                            )?;
+                        }
+                        EndOfEpochTransactionKind::AuthenticatorStateCreate(create) => {
+                            // This can fail if the framework does not yet have the create()
+                            // function.
+                            // TODO: this is an ephemeral constraint - once all production
+                            // networks have the new framework, we can add an .expect() at the
+                            // end here (because at that point this should only be called during
+                            // genesis of new networks).
+                            setup_authenticator_state_create(
+                                create,
+                                temporary_store,
+                                tx_ctx,
+                                move_vm,
+                                gas_charger,
+                                protocol_config,
+                                metrics.clone(),
+                            )?;
+                        }
+                        EndOfEpochTransactionKind::AuthenticatorStateExpire(expire) => {
+                            setup_authenticator_state_expire(
+                                expire,
+                                temporary_store,
+                                tx_ctx,
+                                move_vm,
+                                gas_charger,
+                                protocol_config,
+                                metrics.clone(),
+                            )
+                            .expect("AuthenticatorStateExpire cannot fail");
```

Comment:
Could we tolerate the error here?

Reply:
For now, I don't think so, per our in-person discussion. I have pretty decent tests for the expire function so I think the risk of hitting this is low. I will investigate whether we can safely let it fail despite the shared temporary store. Added a TODO.